packages/Takeover/src/components/FocusTrap.js

```diff
@@ -0,0 +1,36 @@
+import PropTypes from "prop-types";
+import FocusTrapLibrary from "focus-trap-react";
+
+// properties copy from https://github.com/davidtheclark/focus-trap
+
+const propTypes = {
+  /** A function that will be called when the focus trap activates. */
+  onActivate: PropTypes.func,
+  /** A function that will be called when the focus trap deactivates */
+  onDeactivate: PropTypes.func,
+
+  /** By default, when a focus trap is activated the first element in the focus trap's tab order will receive focus. With this option you can specify a different element to receive that initial focus. Can be a DOM node, or a selector string (which will be passed to document.querySelector() to find the DOM node), or a function that returns a DOM node. */
+  initialFocus: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func]),
+
+  /**
+    By default, an error will be thrown if the focus trap contains no elements in its tab order. With this option you can specify a fallback element to programmatically receive focus if no other tabbable elements are found. For example, you may want a popover's <div> to receive focus if the popover's content includes no tabbable elements. Make sure the fallback element has a negative tabindex so it can be programmatically focused. The option value can be a DOM node, a selector string (which will be passed to document.querySelector() to find the DOM node), or a function that returns a DOM node.
+    escapeDeactivates {boolean}: Default: true. If false, the Escape key will not trigger deactivation of the focus trap. This can be useful if you want to force the user to make a decision instead of allowing an easy way out.
+  */
+  fallbackFocus: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func]), //
+
+  /** Default: false. If true, a click outside the focus trap will deactivate the focus trap and allow the click event to do its thing. */
+  clickOutsideDeactivates: PropTypes.bool,
+
+  /** Default: true. If false, when the trap is deactivated, focus will not return to the element that had focus before activation. */
+  returnFocusOnDeactivate: PropTypes.bool,
```

Comment:
Can we leave this out? I think we will _always_ want to do this, and if we really want to focus something else (like a newly created item) it could still be done anyway either in `onAfterClose()` or by a parental component.

Reply:
It doesn't matter if the props declared in this `propTypes` object or nor. It is just for documentation.

If we remove some prop from this list, it can be passed by user anyway and `focus-trap` will change internal behavior.

So removing `returnFocusOnDeactivate` just will make this feature undocumented, but will not disable an ability to use it.

Reply:
yes, let's keep the documentation as it's.